### PR TITLE
Dev 2.3

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**

### CODE REVIEW

File: gui/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+dist/
+.env
```
Comment (jrobinAV) [inline gui/.gitignore]: A new line is needed at the end of a file.

File: .gitignore [hunk 2]
```diff
@@ -6,9 +6,7 @@ __pycache__
 Pipfile.lock
 build/
 *.egg-info/
-
-# mypy
-.mypy_cache/
+dist/
```
Comment (jrobinAV) [inline .gitignore]: Why did you remove `.mypy_cache/` from .gitignore?

File: Pipfile [hunk 3]
```diff
@@ -19,7 +19,7 @@ pytest = "*"
 tox = "*"
 
 [requires]
-python_version = "3.8"
+python_version = "3"
```
Comment (jrobinAV) [inline Pipfile:22]: No opinion on that. What is this python_version used for?

File: gui/dev.env [hunk 4]
```diff
@@ -0,0 +1 @@
+TAIPY_GUI_DIR=<path the taipy package dir parent>
```
Comment (jrobinAV) [inline gui/dev.env]: A new line is needed at the end of a file.

File: .gitignore [hunk 5]
```diff
@@ -6,12 +6,14 @@ __pycache__
...
 
 # IDE settings
 .vscode/
 .idea/
 .idea/taipy.iml
 .DS_Store
+
+.data/
+
+demo-*
```
Comment (jrobinAV) [inline .gitignore]: A new line is needed at the end of a file.

File: demo-scenario_selector.py
Comment (jrobinAV) [inline demo-scenario_selector.py:1]: It should not have been committed? Or it should be in a test folder, or eventually in a demo folder. But not at the root level.

File: .gitignore [hunk 6]
```diff
@@ -6,12 +6,16 @@ __pycache__
...
 .vscode/
 .idea/
 .idea/taipy.iml
 .DS_Store
+
+# Core .data directory
+.data/
+
+# demo files
+demo-*
```
Comment (jrobinAV) [inline .gitignore:23]: It's surprising to me to ignore all files starting with `demo-`. Is it a general rule you have on the GUI repo?

File: gui/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import ScenarioSelector from "./ScenarioSelector";
+
+export {ScenarioSelector}
```
Comment (jrobinAV) [inline gui/src/index.ts]: A new line is needed at the end of a file.

File: src/taipy/gui_core/.gitignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+# This file won't be present on CI and ould prevent the lib to be included if building package locally
+lib/
```
Comment (jrobinAV) [inline src/taipy/gui_core/.gitignore]: A new line is needed at the end of a file.

File: src/taipy/gui_core/__init__.py
Comment (jrobinAV) [inline src/taipy/gui_core/__init__.py:1]: License header missing

**Review by jrobinAV — COMMENTED** [review]
Is it normal to have a gui folder at the root and not under src/ ?
What is the purpose of the GUI folder?

**Review by jrobinAV — COMMENTED** [review]
Is it normal to have a gui folder at the root and not under src/ ?
What is the purpose of the GUI folder?

File: Pipfile (hunk 3 above)
Comment (FredLL-Avaiga) [reply Pipfile:22]: 3.8 would complain if Python 3.8 is unavailable on installation (even if 3.10 is there)

File: demo-scenario_selector.py
Comment (FredLL-Avaiga) [reply demo-scenario_selector.py:1]: Should not be committed

File: .gitignore (hunk 6 above)
Comment (FredLL-Avaiga) [reply .gitignore:23]: yes

**Comment by FredLL-Avaiga** [comment]
> Is it normal to have a gui folder at the root and not under src/ ? What is the purpose of the GUI folder?

yes, this is where we build the js lib (same as in taipy-gui)

File: demo-scenario_selector.py
Comment (FabienLelaquais) [inline demo-scenario_selector.py:1]: Shouldn't this be moved to frontend-experiments?

**Review by FabienLelaquais — APPROVED** [review]
Looks neat!

File: demo-scenario_selector.py
Comment (FredLL-Avaiga) [reply demo-scenario_selector.py:1]: it should

File: src/taipy/gui_core/GuiCoreLib.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+from taipy.core import Core, Scenario
+from taipy.gui import Gui, State
+from taipy.gui.extension import Element, ElementLibrary, ElementProperty, PropertyType
+
+
+class GuiCoreContext:
+    _CORE_CHANGED_NAME = "core_changed"
+    _VAR_NAME = "__CCCtx"
+
+    def __init__(self, gui: Gui, core: Core) -> None:
```
Comment (jrobinAV) [inline src/taipy/gui_core/GuiCoreLib.py:23]: Why do you need a Core instance? It provides no service other than running a dispatcher or managing the versions. I don't see the point of embedding it in a scenario selector.
